spark-connector/v3.4/spark/src/main/java/org/apache/gravitino/spark/connector/hive/SparkHiveTypeConverter34.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.spark.connector.hive;
+
+import org.apache.gravitino.rel.types.Type;
+import org.apache.gravitino.spark.connector.SparkTypeConverter34;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.TimestampType;
+
+public class SparkHiveTypeConverter34 extends SparkTypeConverter34 {
+  @Override
+  public Type toGravitinoType(DataType sparkType) {
+    if (sparkType instanceof TimestampType) {
```

Comment:
throw exception for timestamp_ntz type?

Reply:
why ? Shouldn't it throw an exception for TimestampType with time zone?

Reply:
As commented in https://github.com/apache/gravitino/issues/7046#issuecomment-2834186358, we'd better follow the current Spark Hive behavior, which supports timestamp with timezone only.

Reply:
So you mean when Spark uses timestamp, it automatically converts it to Gravitino `Types.TimestampType.withoutTimeZone();`  when using Spark's timestamp_ntz, an exception is thrown?

Reply:
Following this:

`public class SparkHiveTypeConverter34 extends SparkTypeConverter34 {
  @Override
  public Type toGravitinoType(DataType sparkType) {
    if (sparkType instanceof TimestampType) {
      return Types.TimestampType.withoutTimeZone();
    } else if (sparkType instanceof TimestampNTZType) {
      throw new UnsupportedOperationException("Unsupported TimestampNTZType");
    } else {
      return super.toGravitinoType(sparkType);
    }
  }
}`

Reply:
Yes, we'd better add come comment as the code logic seems odd.